### PR TITLE
feat: add masp transactions data model and migrations

### PR DESCRIPTION
**Pull request by jowparks**
## Summary
Adds MASP transaction db model, migration, client codegen.

Open for debate if this is the best modeling path. I wasn't sure why `Event` is a separate model from `Deposit`, but I followed that pattern.

I also could imagine making these `Send`, `Mint`, and `Burn` separate models, wasn't sure which was preferable. 
## Breaking Change

Is this a breaking change? If yes, add notes below on why this is breaking and
what additional work is required, if any.

```
[ ] Yes
```

### CODE REVIEW

File: prisma/migrations/20221021163739_drop_and_create_index/migration.sql [hunk 1]
```diff
@@ -0,0 +1,5 @@
+-- DropIndex
```
Comment (jowparks) [inline prisma/migrations/20221021163739_drop_and_create_index/migration.sql:1]: This file was autogenerated for me, not sure why, we can delete if needed.

File: package.json [hunk 2]
```diff
@@ -27,8 +27,8 @@
     "test:keepdb": "dotenv -e .env.test -- jest --runInBand",
     "test": "dotenv -e .env.test -- prisma migrate reset --force --skip-generate > /dev/null && yarn test:keepdb",
     "db:client:generate": "prisma generate",
-    "db:migrate": "prisma migrate dev",
-    "db:client:format": "prisma format"
+    "db:client:format": "prisma format",
```
Comment (rohanjadvani) [inline package.json:30]: Should `master` be rebased into this branch?

File: prisma/schema.prisma [hunk 3]
```diff
@@ -229,3 +231,28 @@ model DepositHead {
...
+  block_hash       String              @db.VarChar
+  asset            String              @db.VarChar
+  type             MaspTransactionType
+  block_sequence   Int
+  network_version  Int
+  main             Boolean
+  amount           Int
+  event            Event?
+
+  eventId Int?
```
Comment (rohanjadvani) [inline prisma/schema.prisma:249]:
```suggestion
  event_id Int?
```

Comment (rohanjadvani) [reply prisma/schema.prisma:249]: Will these be unique?

File: prisma/schema.prisma [hunk 4]
```diff
@@ -229,3 +231,28 @@ model DepositHead {
...
   @@map("deposit_head")
 }
+
+model MaspTransaction {
+  id               Int                 @id @default(autoincrement())
+  created_at       DateTime            @default(now()) @db.Timestamp(6)
+  updated_at       DateTime            @default(now()) @updatedAt @db.Timestamp(6)
+  transaction_hash String              @db.VarChar
+  block_hash       String              @db.VarChar
+  asset            String              @db.VarChar
```
Comment (rohanjadvani) [inline prisma/schema.prisma:241]: We've been using asset identifier in the MASP prototypes. What do you think about that?

File: prisma/schema.prisma [hunk 5]
```diff
@@ -229,3 +231,28 @@ model DepositHead {
...
+  amount           Int
+  event            Event?
+
+  eventId Int?
+
+  @@map("masp_transactions")
+}
+
+enum MaspTransactionType {
+  SEND
```
Comment (rohanjadvani) [inline prisma/schema.prisma:255]: What do you think about `TRANSFER`? I know we had `SEND` in the spec, but we've been using that vocabulary in the MASP prototypes

File: package.json (hunk 2 above)
Comment (jowparks) [reply package.json:30]: I already rebased, I think I flipped order is all (had created PR for format after this one).

File: prisma/schema.prisma (hunk 3 above)
Comment (jowparks) [reply prisma/schema.prisma:249]: I am not sure for these two questions, I was imagining the event represents a sent event from client. Not sure if we want to dedupe and prevent secondary save or if we want to save both and have one to many for masp <- event. I think you guys would be better at making the decision on that given you have been through a couple testnet phases already

File: prisma/schema.prisma (hunk 4 above)
Comment (jowparks) [reply prisma/schema.prisma:241]: SGTM

File: prisma/schema.prisma [hunk 6]
```diff
@@ -229,3 +231,28 @@ model DepositHead {
...
   @@map("deposit_head")
 }
+
+model MaspTransaction {
+  id               Int                 @id @default(autoincrement())
+  created_at       DateTime            @default(now()) @db.Timestamp(6)
+  updated_at       DateTime            @default(now()) @updatedAt @db.Timestamp(6)
+  transaction_hash String              @db.VarChar
+  block_hash       String              @db.VarChar
+  asset_identifier String              @db.VarChar
```
Comment (jowparks) [inline prisma/schema.prisma:241]: I was guessing this identifier is a hash, so I put as string. Wondering if we want any other attributes of the assets/transactions and if we want to index on the identifier.

File: prisma/schema.prisma [hunk 7]
```diff
@@ -229,3 +231,28 @@ model DepositHead {
...
+  type             MaspTransactionType
+  block_sequence   Int
+  network_version  Int
+  main             Boolean
+  amount           Int
+  event            Event?
+
+  event_id Int?
```
Comment (rohanjadvani) [inline prisma/schema.prisma:249]: I don't think this `event_id` is necessary. I imagine this should look fairly similar to `Deposit`

File: prisma/schema.prisma (hunk 6 above)
Comment (rohanjadvani) [reply prisma/schema.prisma:241]: I don't think we need to search MASP transactions based on a given asset yet, so I think we can omit for now.

File: prisma/migrations/20221031181536_add_masp_transactions/migration.sql [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+ALTER TABLE "events" ADD COLUMN     "masp_transaction_id" INTEGER;
+
+-- CreateTable
+CREATE TABLE "masp_transactions" (
+    "id" SERIAL NOT NULL,
+    "created_at" TIMESTAMP(6) NOT NULL DEFAULT CURRENT_TIMESTAMP,
+    "updated_at" TIMESTAMP(6) NOT NULL DEFAULT CURRENT_TIMESTAMP,
+    "transaction_hash" VARCHAR NOT NULL,
+    "block_hash" VARCHAR NOT NULL,
+    "asset" VARCHAR NOT NULL,
```
[the remark this answers is not in the document]
Comment (jowparks) [reply prisma/migrations/20221031181536_add_masp_transactions/migration.sql]: Yup you are right, deleted the migration and regenerated.

**Review by rohanjadvani — APPROVED** [review]

**Pull request merged**